include/RAJA/pattern/teams.hpp

```diff
@@ -0,0 +1,40 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing user interface for RAJA::kernel
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_teams_HPP
+#define RAJA_pattern_teams_HPP
+
+#include "RAJA/pattern/teams/teams_core.hpp"
+
+//
+// All platforms must support loop execution.
+//
+#include "RAJA/pattern/teams/teams_sequential.hpp"
+
+#if (defined(__NVCC__) || (defined(__clang__) && defined(__CUDA__))) && defined(RAJA_ENABLE_CUDA)
```

Comment:
@tomstitt I think these guards should address our needs.

Reply:
There are still issues in *teams_core.hpp*; `RAJA_ENABLE_DEVICE` is based on only `RAJA_ENABLE_CUDA` and is then used in `LaunchPolicy` and `LoopPolicy`, meaning you need to provide a `DEVICE_POLICY` but there are no device policies since we aren't using a device compiler. We'll either need a dummy policy or to guard `RAJA_ENABLE_DEVICE` more strongly

Reply:
Pushed a fix. I don't think ``RAJA_DEVICE_CODE`` would work since that only detects if you are inside a kernel.